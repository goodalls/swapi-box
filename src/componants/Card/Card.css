.card {
  border: 2px solid yellow;
  border-radius: 25px;
  padding: 20px 10px;
  background-color: black;
  box-shadow: 2px 2px 4px #868686;
  min-height: 250px;
  margin: 20px; }

.card:hover {
  cursor: pointer;
  background-color: yellow; }

.card p {
  color: yellow; }
