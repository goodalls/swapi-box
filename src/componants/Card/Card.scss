.card {
  background-color: black;
  border-radius: 25px;
  border: 2px solid yellow;
  box-shadow: 2px 2px 4px #868686;
  margin: 20px;
  padding: 20px 10px;
  text-align: center;
  display: grid;
}

.card:hover {
  cursor: pointer;
  background-color: grey;
}

.card p {
  color: yellow;
  margin: 5%;
}

.card button {
  background-color: black;
  color: yellow;
}

.card .favorited{
  background-color: yellow;
  color: black;
}