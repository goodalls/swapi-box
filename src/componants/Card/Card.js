import React from 'react';
import './Card.css';

const Card = () => {
  return (
    <div className="Card">
      <p>Name</p>
      <p>Homeworld: Population</p>
      <p>Species</p>
      <button>favorite</button>
    </div>
  );
};

export default Card;
