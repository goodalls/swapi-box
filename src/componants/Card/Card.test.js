import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let mockFav;
  let mockFavArray;
  let wrapper;

  beforeEach(() => {
    mockFav = {name: 'george'};
    mockFavArray = [{name: 'jefferson'}];
    wrapper = shallow(<Card favorite={mockFav} favorites={mockFavArray}/>);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should add favorited to class', () => {
    mockFavArray = [{ name: 'george' }];
    expect(wrapper).toMatchSnapshot();
  });
});
