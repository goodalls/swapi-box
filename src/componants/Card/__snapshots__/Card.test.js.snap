// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should add favorited to class 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorite={
        Object {
            "name": "george",
          }
    }
    favorites={
        Array [
            Object {
              "name": "jefferson",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorite={
        Object {
            "name": "george",
          }
    }
    favorites={
        Array [
            Object {
              "name": "jefferson",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot of people card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorite={
        Object {
            "name": "george",
          }
    }
    favorites={
        Array [
            Object {
              "name": "jefferson",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot of planet card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorite={
        Object {
            "name": "george",
          }
    }
    favorites={
        Array [
            Object {
              "name": "jefferson",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot of vehicle card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorite={
        Object {
            "name": "george",
          }
    }
    favorites={
        Array [
            Object {
              "name": "jefferson",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
